### PR TITLE
[terminal] register CTRL+* keybindings to CTRL on OSX (fixes #907)

### LINKED ISSUE #907: Terminal has unconsumed 'passthrough' keybindings
CTRL+P (find file) doesn't work in the terminal context, but also doesn't seem to do anything and doesn't prevent default. As a result the browser's print dialog opens up. (cc @simark)

### PR DESCRIPTION
**Pull request by svenefftinge**
Signed-off-by: Sven Efftinge <sven.efftinge@typefox.io>

### CODE REVIEW

File: packages/terminal/src/browser/terminal-frontend-contribution.ts [hunk 1]
```diff
@@ -83,9 +84,14 @@ export class TerminalFrontendContribution implements CommandContribution, MenuCo
         /* Register ctrl + k (the passed Key) as a passthrough command in the
            context of the terminal.  */
         const regCtrl = (k: Key) => {
+            let modifier = isOSX ? Modifier.M4 : Modifier.M1;
+            if (Key.KEY_A.keyCode === k.keyCode) {
```
Comment (simark) [inline packages/terminal/src/browser/terminal-frontend-contribution.ts]: With this, ctrl-A will not be registered as a passthrough keybinding on Mac.  isn't it wrong?  Don't you want to register cmd-A in addition to ctrl-A?

In the xterm.js code, ctrl-A is handled with the rest (it sends a 0x01 byte), and cmd-A is handled in a special way for Mac:
https://github.com/xtermjs/xterm.js/blob/v3/src/Terminal.ts#L1684

**Comment by kittaakos** [comment]
I am verifying it now.

**Comment by kittaakos** [comment]
On Mac, Ctrl+A selects the empty lines too. That means, even if I have only 5-6 lines in the terminal, I get all the blank lines copied into my clipboard after doing Ctrl+A, Ctrl+C.

**Comment by svenefftinge** [comment]
> On Mac, Ctrl+A selects the empty lines too. That means, even if I have only 5-6 lines in the terminal, I get all the blank lines copied into my clipboard after doing Ctrl+A, Ctrl+C.

Did you mean CMD+A? It is the behavior of xterm.js and for me it is the same in native osx terminals.

**Comment by kittaakos** [comment]
Cmd+K does not clear the screen.

**Comment by svenefftinge** [comment]
Please compare with osx terminal

**Comment by kittaakos** [comment]
> Cmd+K does not clear the screen.

Ctrl+L does that, so we can leave with this.

**Comment by kittaakos** [comment]
> Did you mean CMD+A?

Yes, I meant Cmd+A.

> same in native osx terminals

Not for me. OK.

**Comment by simark** [comment]
In [xterm.js](https://github.com/xtermjs/xterm.js/blob/4f144dc6eb8bdb56dd4d705ca356694248ac2acc/src/Terminal.ts#L1706), the alt+<letter> bindings are also not used on Mac:

```
        } else if (!this.browser.isMac && ev.altKey && !ev.ctrlKey && !ev.metaKey) {
          // On Mac this is a third level shift. Use <Esc> instead.
          if (ev.keyCode >= 65 && ev.keyCode <= 90) {
            result.key = C0.ESC + String.fromCharCode(ev.keyCode + 32);
          } else if (ev.keyCode === 192) {
            result.key = C0.ESC + '`';
          } else if (ev.keyCode >= 48 && ev.keyCode <= 57) {
            result.key = C0.ESC + (ev.keyCode - 48);
          }
} 
```

Do you know what they mean about the third level shift?

**Comment by svenefftinge** [comment]
No, I didn't understand it, but I tested the `alt+*`keybindings and they seem to do the same in theia and in my terminal.

**Pull request merged**